index.js

```diff
@@ -682,10 +682,29 @@
   //. > S.maybe(0, R.length, S.Nothing())
   //. 0
   //. ```
-  S.maybe = def('maybe', [a, Function, Maybe], function(x, f, maybe) {
+  var maybe = S.maybe =
+  def('maybe', [a, Function, Maybe], function(x, f, maybe) {
     return fromMaybe(x, maybe.map(f));
   });
 
+  //  catMaybes :: [Maybe a] -> [a]
+  var catMaybes = def('catMaybes', [Accessible], R.chain(maybe([], R.of)));
+
+  //# mapMaybe :: (a -> Maybe b) -> [a] -> [b]
+  //.
+  //. Takes a function and a list, and returns a list containing the value
+  //. of each Just resulting from the application of the function to each
+  //. element of the input list.
+  //.
```

Comment:
I, too, like the description from the Haskell library. The first sentence sets the scene nicely. The main problem I have with using this definition verbatim is that it's not consistent with the form used to describe other Sanctuary functions: _Takes a foo and a bar, and returns a baz._ I intended the second paragraph to give the high-level description:

```javascript
//. In general terms, `mapMaybe` filters a list while mapping over it.
```

I have one other problem with the description from the Haskell library (a minor one).

> If this is `Nothing`, no element is added on to the result list.

I don't like the use of "added on to", as it describes an implementation detail not relevant to the caller. It's nice to ignore the intermediate states between input and output. Functions are applied atomically in mathematics, and we can think of the application of a pure JavaScript functions in this way, even though we know it to be untrue.